File: repro/cmake-mpi-omp/hello.c
#include <mpi.h>
#include <omp.h>
#include <stdio.h>

int main(int argc, char **argv)
{
  int rank = 0;
  MPI_Init(&argc, &argv);
  MPI_Comm_rank(MPI_COMM_WORLD, &rank);

#pragma omp parallel
  {
    if (rank == 0 && omp_get_thread_num() == 0) {
      printf("Hello from C + OpenMP + MPI\n");
    }
  }

  MPI_Finalize();
  return 0;
}

File: repro/cmake-mpi-omp/hello.f90
program hello_mpi_omp
  use mpi
  implicit none
  integer :: ierr, rank

  call MPI_Init(ierr)
  call MPI_Comm_rank(MPI_COMM_WORLD, rank, ierr)

!$omp parallel
  if (rank == 0) then
    print *, 'Hello from Fortran + OpenMP + MPI'
  end if
!$omp end parallel

  call MPI_Finalize(ierr)
end program hello_mpi_omp
